pallets/loans-ref/src/migrations/v1.rs

```diff
@@ -0,0 +1,133 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+	pallet_prelude::*, storage::bounded_vec::BoundedVec, storage_alias, traits::OnRuntimeUpgrade,
+	weights::Weight, Blake2_128Concat, RuntimeDebug,
+};
+use scale_info::TypeInfo;
+use sp_std::vec::Vec;
+
+use crate::{
+	write_off::{WriteOffRule, WriteOffTrigger},
+	*,
+};
+
+mod v0 {
+	use super::*;
+
+	#[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
+	pub struct WriteOffState<Rate> {
+		pub overdue_days: u32,
+		pub percentage: Rate,
+		pub penalty: Rate,
+	}
+
+	pub type WriteOffStates<T> =
+		BoundedVec<v0::WriteOffState<<T as Config>::Rate>, <T as Config>::MaxWriteOffPolicySize>;
+
+	#[storage_alias]
+	pub(crate) type WriteOffPolicy<T: Config> =
+		StorageMap<Pallet<T>, Blake2_128Concat, PoolIdOf<T>, WriteOffStates<T>, ValueQuery>;
+}
+
+/// This updates the policy to the newer version.
+pub struct Migration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for Migration<T> {
+	fn on_runtime_upgrade() -> Weight {
+		if Pallet::<T>::on_chain_storage_version() > StorageVersion::new(0) {
+			log::warn!("Migration was already done. This migration can be removed");
+			return Weight::zero();
+		}
+
+		let mut count = 0;
+		WriteOffPolicy::<T>::translate_values(|policy: v0::WriteOffStates<T>| {
+			count += 1;
+			Some(
+				policy
+					.into_iter()
+					.map(|old| {
+						WriteOffRule::new(
+							[WriteOffTrigger::PrincipalOverdueDays(old.overdue_days)],
+							old.percentage,
+							old.penalty,
+						)
+					})
+					.collect::<Vec<_>>()
+					.try_into()
+					.expect("Size of the new vec can not be longer than previous one, qed"),
+			)
+		});
+
+		Pallet::<T>::current_storage_version().put::<Pallet<T>>();
+
+		log::info!("Successful migration: v0 -> v1. Items: {count}");
+
+		T::DbWeight::get().reads_writes(count, count)
```

Comment:
Nit: It's actually `count + 1` since we read and write the `StorageVersion`:

```suggestion
		T::DbWeight::get().reads_writes(count.saturating_add(1), count.saturating_add(1))
```

You could also initialize `count = 1` and log 

```
		log::info!("Successful migration: v0 -> v1. Items: {count.saturating_sub(1)}");

```

Reply:
I thought the same at first, but then I checked some substrate pallets, and their Substrate returned `Weight::zero()` after reading the `StorageVersion`, which maybe means the version is something stored in the memory, not in the database 🤔.

I've tried to check if I'm right, but it's done procedurally...

Reply:
Anyways adding 1 does not hurt!